azext_iot/tests/test_iot_pnp_int.py

```diff
@@ -0,0 +1,183 @@
+        assert json.dumps(interface)
+        assert interface['@id'] == self.kwargs['interface_id']
+        assert interface['displayName'] == 'MXChip1'
+        assert len(interface['contents']) > 0
+
+        # Success: Update Interface
+        interface = str(read_file_content(self.kwargs['interface']))
+        display_name = json.loads(interface)['displayName']
+        interface_newContent = interface.replace(display_name, '{}-Updated'.format(display_name))
+        interface_newContent = interface_newContent.replace('\n', '')
+        fo = open(self.kwargs['interface-updated'], "w+", encoding='utf-8')
+        fo.write(interface_newContent)
+        fo.close()
+        self.cmd('iot pnp interface update -e {endpoint} -r {repo} --def {interface-updated}', checks=self.is_empty())
+
+        # Todo: Publish Interface
+
+        # Success: Delete Interface
+        self.cmd('iot pnp interface delete -e {endpoint} -r {repo} -i {interface_id}', checks=self.is_empty())
+
+    def test_model_life_cycle(self):
+
+        # Checking the Capability-Model list
+        self.cmd('iot pnp capability-model list -e {endpoint} -r {repo}',
+                 checks=[self.check('length([*])', 0)])
+
+        # Error: missing repo-id or login
+        self.cmd('iot pnp capability-model create -e {endpoint} --def {model}', expect_failure=True)
+
+        # Error: Invalid Capability-Model definition file
+        self.cmd('iot pnp capability-model create -e {endpoint} -r {repo} --def model', expect_failure=True)
+
+        # Error: wrong path of Capability-Model definition
+        self.cmd('iot pnp capability-model create -e {endpoint} -r {repo} --def model.json', expect_failure=True)
+
+        # Success: Create new Capability-Model
+        self.cmd('iot pnp capability-model create -e {endpoint} -r {repo} --def {model}', checks=self.is_empty())
+
+        # Checking the Capability-Model list
+        self.cmd('iot pnp capability-model list -e {endpoint} -r {repo}',
+                 checks=[self.check('length([*])', 1)])
+
+        # Get Capability-Model
+        model = self.cmd('iot pnp capability-model show -e {endpoint} -r {repo} -m {model_id}').get_output_in_json()
+        assert json.dumps(model)
+        assert model['@id'] == self.kwargs['model_id']
+        assert len(model['implements']) > 0
+
+        # Success: Update Capability-Model
+        model = str(read_file_content(self.kwargs['model']))
+        display_name = json.loads(model)['displayName']
+        model_newContent = model.replace(display_name, '{}-Updated'.format(display_name))
+        model_newContent = model_newContent.replace('\n', '')
+        fo = open(self.kwargs['model-updated'], "w+", encoding='utf-8')
+        fo.write(model_newContent)
+        fo.close()
+        self.cmd('iot pnp capability-model update -e {endpoint} -r {repo} --def {model-updated}', checks=self.is_empty())
+
+        # Todo: Publish Capability-Model
```

Comment:
Please add a test for capability model Publish.

Reply:
Added